src/systemTest/java/com/flightstats/hub/system/functional/ChannelMaxItemsEnforcementTest.java

```diff
@@ -38,60 +39,87 @@
-                .until(() -> s3Service.confirmItemsInS3(type, items, channelName) &&
-                        items.stream().allMatch(path -> channelItemRetriever.getItem(path).isPresent()));
+                .until(() -> {
+                    if (type.equals(ChannelType.BOTH)) {
+                        return items.stream().allMatch(item -> confirmInLocation(item, Location.LONG_TERM_SINGLE)) &&
+                                items.stream().allMatch(item -> confirmInLocation(item, Location.LONG_TERM_BATCH));
+                    } else {
+                        Location location = type.equals(ChannelType.BATCH) ? Location.LONG_TERM_BATCH : Location.LONG_TERM_SINGLE;
+                        return items.stream().allMatch(item -> confirmInLocation(item, location));
+                    }
+                });
     }
 
-    private String setupTest(ChannelType type) {
+    private void createChannelWithMaxItems(ChannelType type) {
         channelName = randomAlphaNumeric(10);
         ChannelConfig channelConfig = ChannelConfig.builder()
                 .name(channelName)
                 .storage(type.name())
-                .maxItems(5)
+                .maxItems(MAX_ITEMS)
                 .build();
         channelConfigService.create(channelConfig);
+    }
 
-        addItemsOverTime();
-        String path = items.get(0);
-        waitUntilAllItemsAreInS3(type);
-        return path;
+    private void addMaxItems() {
+        items.addAll(channelItemCreator.addItems(channelName, TEST_DATA, MAX_ITEMS));
+    }
+
+    @SneakyThrows
+    private void waitAMinute(ChannelType type) {
+        /*
+            batch items are compared with granularity of one minute
+            so if the excess items are of the same minute they will not be deleted
+            TODO: a different solution and a fix for the root cause
+        */
+        if (!type.equals(ChannelType.SINGLE)) {
+            Thread.sleep(TimeUnit.MINUTES.toMillis(1));
+        }
+    }
+
+    private String addExtraItem() {
+        String item = channelItemCreator.addItem(channelName, TEST_DATA);
+        items.add(item);
+        return item;
     }
 
     private void confirmDelete(ChannelType type, String path) {
         Awaitility.await().pollInterval(Duration.FIVE_SECONDS)
                 .atMost(Duration.TWO_MINUTES)
-                .until(() -> s3Service.confirmItemNotInS3(type, path, channelName));
+                .until(() -> {
+                    if (type.equals(ChannelType.BOTH)) {
+                        return !confirmInLocation(path, Location.LONG_TERM_SINGLE) &&
+                                !confirmInLocation(path, Location.LONG_TERM_BATCH);
+                    } else {
+                        Location location = type.equals(ChannelType.BATCH) ? Location.LONG_TERM_BATCH : Location.LONG_TERM_SINGLE;
+                        return !confirmInLocation(path, location);
+                    }
+                });
```

Comment:
Is this asserting stuff? I think you can add a second param to `.until` that takes a hamcrest matcher and then it would be.